Firestore/core/src/firebase/firestore/core/field_filter.h

```diff
@@ -29,29 +30,26 @@ namespace firestore {
 namespace core {
 
 /**
- * RelationFilter is a document filter constraint on a query with a single
+ * FieldFilter is a document filter constraint on a query with a single
  * relation operator.
  */
-class RelationFilter : public Filter {
+class FieldFilter : public Filter {
  public:
-  RelationFilter() = default;
-
   /**
-   * Creates a new filter that compares fields and values. Only intended to be
-   * called from Filter::Create().
-   *
-   * @param field A path to a field in the document to filter on. The LHS of the
-   *     expression.
-   * @param op The binary operator to apply.
-   * @param value_rhs A constant value to compare @a field to. The RHS of the
-   *     expression.
+   * Creates a Filter instance for the provided path, operator, and value.
    */
-  RelationFilter(model::FieldPath field,
-                 Operator op,
-                 model::FieldValue value_rhs);
+  static std::shared_ptr<FieldFilter> Create(model::FieldPath path,
+                                             Operator op,
+                                             model::FieldValue value_rhs);
+
+  FieldFilter() = default;
```

Comment:
Can this constructor be avoided? It looks like it could be error-prone.

Reply:
It was useful when these were embedded in their Objective-C equivalents. Removed.